src/core/packages/elasticsearch/server-internal/src/version_check/ensure_es_version.ts

```diff
@@ -174,22 +179,41 @@ export const pollEsNodesVersion = ({
 
   const isStartup$ = new BehaviorSubject(hasStartupInterval);
 
+  let currentInterval = 0;
   const checkInterval$ = isStartup$.pipe(
     distinctUntilChanged(),
     map((useStartupInterval) =>
       useStartupInterval ? healthCheckStartupInterval! : healthCheckInterval
-    )
+    ),
+    tap((ms) => (currentInterval = ms))
   );
 
-  return checkInterval$.pipe(
-    switchMap((checkInterval) => interval(checkInterval)),
-    startWith(0),
+  const tick$ = checkInterval$.pipe(
+    switchMap((checkInterval) => interval(checkInterval).pipe(startWith(0))),
+    shareReplay({ refCount: true })
+  );
+
+  return tick$.pipe(
```

Comment:
I think we can remove `tick$` now that we are not using it in delay (did not test locally)


```suggestion
);

return checkInterval$.pipe(
    switchMap((checkInterval) => interval(checkInterval)),
    startWith(0),
```

Reply:
Great idea! I tested and it works 🥳